frontend/catalyst/jit.py

```diff
@@ -430,6 +433,33 @@ def sum_abstracted(arr):
     if fn is None:
         return functools.partial(qjit, **kwargs)
 
+    # Map static_argnames to static_argnums
+    # no need to propagate static_argnames further
+    if static_argnames is not None:
```

Comment:
I think my preference would be to handle this kwarg in the same place that we handle other options, so as not to have inconsistent handling of options by having the code in different places.

Reply:
The problem is, the jitted function is not available inside the `CompileOption` class, so this map won't be possible there.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1158/commits/bf05ee6eeaaaf9e9f8d81c75f5ac2f0f7b367851 is a workaround but I think this is more ugly

Reply:
I am now merging static_argname into static_argnum in the QJIT class's __init__.